### PR TITLE
Lint with black and set up automated black check

### PR DESCRIPTION
**Pull request by mattwthompson**
Split out from #650. `isort` and `black` have some small disagreements that could be resolved here but I would rather deal with those later while resolving some `flake8` complaints. The changes here are roughly the minimum to get this adopted as quickly as possible, and I tried to keep the commit history discrete if we wish to roll anything back.

- [x] Tag issue being addressed
- [ ] Add [tests](https://github.com/openforcefield/openforcefield/tree/master/openforcefield/tests)
- [x] Update docstrings/[documentation](https://github.com/openforcefield/openforcefield/tree/master/docs), if applicable
- [ ] Update [changelog](https://github.com/openforcefield/openforcefield/blob/master/docs/releasehistory.rst)

### CODE REVIEW

File: docs/developing.rst [hunk 1]
```diff
@@ -266,11 +266,16 @@ The naming conventions of classes, functions, and variables follows `PEP8 <https
...
-Anything not covered above is up to personal preference.
-To remove the human friction from code formatting, we will likely adopt a standard formatter like `black <https://github.com/psf/black>`_ in the near future.
+There is a step in CI that uses these tool(s) to check for a consistent style. To ensure that changes follow these standards, you can install and run these tool(s) locally:
+
+.. code-block:: bash
 
+    $ conda install black -c conda-forge
+    $ black openforcefield
+
+Anything not covered above is up to personal preference.
```
Comment (SimonBoothroyd) [inline docs/developing.rst:281]:
```suggestion
Anything not covered above is currently up to personal preference, but may change as new linters are added.
```

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+name: lint
+
+on:
+  push:
+    paths:
+      - 'openforcefield/**.py'
+      - 'setup.cfg'
```
Comment (SimonBoothroyd) [inline .github/workflows/lint.yml:7]:
```suggestion
```

Can probably drop these lines as the linting step is normally very fast.

**Comment by mattwthompson** [comment]
Thanks @SimonBoothroyd!

**Comment by j-wags** [comment]
Looks great. Thanks for making the call and merging this in -- I've been hesitating too much on this and it's good to just drop the hammer and deal with the consequences instead of leaving it hanging forever!

**Comment by j-wags** [comment]
Looks great. Thanks for making the call and merging this in -- I've been hesitating too much on this and it's good to just drop the hammer and deal with the consequences instead of leaving it hanging forever!

**Pull request merged**